java/src/main/java/io/opentracing/grpc/ClientTracingInterceptor.java

```diff
@@ -0,0 +1,290 @@
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientCall.Listener;
+import io.grpc.ClientInterceptor;
+import io.grpc.ClientInterceptors;
+import io.grpc.Context;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+
+import io.opentracing.contrib.grpc.OpenTracingContextKey;
+import io.opentracing.contrib.grpc.ClientRequestAttribute;
+import io.opentracing.propagation.TextMap;
+import io.opentracing.propagation.Format;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/** 
+ * An interceptor that applies tracing via OpenTracing to all client requests. 
+ */ 
+public class ClientTracingInterceptor implements ClientInterceptor {
+    
+    private final Tracer tracer;
+    private final String operationName;
+    private final boolean streaming;
+    private final boolean verbose;
+    private final Set<ClientRequestAttribute> tracedAttributes;
+
+    /**
+     * @param tracer to use to trace requests
+     */
+    public ClientTracingInterceptor(Tracer tracer) {
+        this.tracer = tracer;
+        this.operationName = "";
+        this.streaming = false;
+        this.verbose = false;
+        this.tracedAttributes = new HashSet<ClientRequestAttribute>();
+    }
+
+    private ClientTracingInterceptor(Tracer tracer, String operationName, boolean streaming,
+        boolean verbose, Set<ClientRequestAttribute> tracedAttributes) {
+        this.tracer = tracer;
+        this.operationName = operationName;
+        this.streaming = streaming;
+        this.verbose = verbose;
+        this.tracedAttributes = tracedAttributes;
+    }
+
+    /**
+     * Use this interceptor to trace all requests made by this client channel.
+     * @param channel to be traced
+     */ 
+    public Channel intercept(Channel channel) {
+        return ClientInterceptors.intercept(channel, this);
+    }
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+        MethodDescriptor<ReqT, RespT> method, 
+        CallOptions callOptions, 
+        Channel next
+    ) {
+        final String operationName;
+        if (this.operationName.equals("")) {
+            operationName = method.getFullMethodName();
+        } else {
+            operationName = this.operationName;
+        }
+
+        Span activeSpan = OpenTracingContextKey.activeSpan();
+        final Span span = createSpanFromParent(activeSpan, operationName);
+        System.out.println("Started span " + span.toString());
+
+        for (ClientRequestAttribute attr : this.tracedAttributes) {
+            switch (attr) {
+                case AFFINITY:
+                    if (callOptions.getAffinity() == null) {
+                        span.setTag("Affinity", "null");
```

Comment:
let's prefix all of these with `"grpc."` and use underscores for word boundaries a la http://opentracing.io/data-semantics/#http-server-tags